Dashboard/srushti

srushtisalke opened the pull request.

srushtisalke committed:
Created styling for dashboard.
Added required classes in some components and pages for having classes and subclasses required in css styling.
Created right and left sections for dashboard.
Updated files as required.

srushtisalke commented:
Please find video link here:
https://www.loom.com/share/3441b0e29bef4ff79f8778710f01ba74?sid=ed2f4475-c72c-464d-9d78-163d4ae97872

srushtisalke commented:
@aina21 could you please review it ?

aina21 commented:
Please make sure to resolve the existing conflicts.

aina21 commented on src/styles/teamDetails.css:
```diff
@@ -0,0 +1,27 @@
+.Team-Member-container {
+  /* background-color: aqua; */
```
remove it

srushtisalke replied:
done

aina21 commented on src/styles/team.css:
```diff
@@ -0,0 +1,58 @@
+  background-color:  #ffb98a;
+}
+.container-heading .logo{
+  /* padding-left: 4em; */
```
remove it

aina21 commented on src/pages/Team.jsx:
```diff
@@ -1,57 +1,40 @@
+return(
+  <>
+  <section className="welcome-team">
+    {/* <h1>Welcom Back, {team.team_name}</h1> */}
```
remove it

srushtisalke replied:
done

aina21 commented on src/pages/Team.jsx:
```diff
@@ -1,57 +1,40 @@
+  <>
+  <section className="welcome-team">
+    {/* <h1>Welcom Back, {team.team_name}</h1> */}
+    <MembersList></MembersList>
```
`<MembersList/>`

srushtisalke replied:
done

aina21 commented on src/pages/Team.jsx:
```diff
@@ -1,57 +1,40 @@
+  <div className="container-section">
+    <section className="left-section">
+      <section className="TeamMember">
+        <TeamMembers></TeamMembers>
```
same here, if you don't need to include anything between the opening and closing tags of a component, you can use a self-closing tag instead.

srushtisalke replied:
done

srushtisalke committed:
Resolved conflicts in following files:

# Conflicts:
#	src/component/CreateMember.jsx
#	src/component/CreateTimeoff.jsx
#	src/component/TeamDetails.jsx
#	src/pages/Team.jsx

Added functionality to fetch Team name on dashboard.
Updates as reviewer suggested.

srushtisalke commented:
@aina21  I have done changes and resolved conflicts as well. 
Please find all changes and request in commit section.

srushtisalke committed:
Fixed for fetching team name

aina21 commented on src/component/MembersList.jsx:
```diff
@@ -1,10 +1,14 @@
+  const [team, setTeam] = useState(null);
+  const { code } = useParams();
 
   useEffect(() => {
```
as we fetch data for both team and member in the useEffect, try to use promise.all if possible to improve effeciency. something like

```js
const [teamData, membersData] = await Promise.all([
          fetchTeam(),
          fetchMembers()
        ]);
```

aina21 commented on src/component/MembersList.jsx:
```diff
@@ -27,7 +48,15 @@ const MembersList = ({teamId}) => {
     <div className="containerList">
       <div className="titleContainer">
-        <h4>Here is a list of your team members!</h4>
+        <section>
```
Shouldn’t the condition also include the `<section>` element itself? If the team does not exist, the `<section>` element will still be added to the DOM without any value, which might not be ideal.

srushtisalke committed:
update changes as per reviewers
